core/controllers/creator_dashboard_test.py

```diff
@@ -644,6 +679,77 @@ def test_can_see_subscribers(self):
         response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
         self.assertEqual(len(response['subscribers_list']), 0)
 
+    def test_get_topic_summary_dicts(self):
+        self.login(self.OWNER_EMAIL)
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertEqual(len(response['topic_summary_dicts']), 0)
+            self.save_new_topic(
+                'topic_id', self.owner_id, 'Name', 'Description',
+                ['story_id_1', 'story_id_2'], ['story_id_3'],
+                ['skill_id_1', 'skill_id_2'], [], 1)
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertEqual(len(response['topic_summary_dicts']), 1)
+            self.assertTrue(isinstance(response['topic_summary_dicts'], list))
+        self.logout()
+
+    def test_can_update_display_preference(self):
+        self.login(self.OWNER_EMAIL)
+        display_preference = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['display_preference']
+        self.assertEqual(display_preference, 'card')
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL,
+            {'display_preference': 'list'},
+            csrf_token=csrf_token)
+        display_preference = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['display_preference']
+        self.assertEqual(display_preference, 'list')
+        self.logout()
+
+    def test_get_collections_list(self):
+        self.set_admins([self.OWNER_USERNAME])
+        self.login(self.OWNER_EMAIL)
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        collection_list = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['collections_list']
+        self.assertEqual(collection_list, [])
+        collection_id = self.post_json(
+            feconf.NEW_COLLECTION_URL, {}, csrf_token=csrf_token)[
+                creator_dashboard.COLLECTION_ID_KEY]
+        collection_list = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['collections_list']
+        self.assertEqual(collection_list[-1]['id'], collection_id)
+        self.assertEqual(len(collection_list), 1)
```

Comment:
Well this creates a default collection with title, objective all equal to ''. This is why I asserted the length instead of the data as there is no appropriate data to be tested. The rest of the fields like id are random.

Reply:
Then update the collection to actually have useful data, and test that. You already have the collection_id in the test and you can use that as the expected value for the test.

Part of the point of the test is to make sure that the correct data is returned, not just any data. There should be no room for error.

Reply:
I already tested the id

Reply:
Ah, I was referring to your comment "The rest of the fields like id are random".

My earlier suggestion about updating the collection to have useful data and testing the equality of the `collection_list` variable against the desired value still stands.

Reply:
Done